File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int maxLen = 0, l = 0, r = 1;
        Set<Character> seen = new HashSet<>();

        while (r < s.length()){
            char c = s.charAt(r-1);

            if (seen.contains(c)){
                seen.remove(c);
                l++;
            }
            else {
                seen.add(c);
                r++;
                maxLen = Math.max(maxLen, seen.size());
            }
        }
        return maxLen;

    }
}

File: Data Structures & Algorithms/sqrtx/submission-1.py
class Solution:
    def mySqrt(self, x: int) -> int:
        l, r = 0, math.ceil(x / 2)
        res = 0

        while l <= r:
            m = l + (r - l) // 2

            if m * m <= x:
                res = m
                l = m + 1
            else:
                r = m - 1
        return res